designdocs/batched_messages.md

```diff
@@ -0,0 +1,96 @@
+# Batched Messages
+
+## Table of Contents
+* [Context and Scope](#context-and-scope)
+* [Problem Statement](#problem-statement)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Proposal](#proposal)
+* [Benefits and Risks](#benefits-and-risks)
+* [Alternatives and Rationale](#alternatives-and-rationale)
+* [Glossary](#glossary)
+
+## Context and Scope
+This design document describes batched messages and APIs for sending and retrieving them.
+
+## Problem Statement
+In order to reduce costs for announcers of messages on-chain as well as reduce network congestion, announcers collate messages into batches of the same type of message and announce the batch location on-chain, instead of announcing each individual message.
+
+## Goals and Non-Goals
+This specifies how batches are to be announced on chain: what is required and how a batch may be partially verified based on on-chain information.
+
+This document does not describe the messages that are supported for batch announcements. Please see the [DSNP Specification](https://spec.dsnp.org/).
+
+This document does not discuss validation of batch file format located at the URL in the announcement, since
+the file format cannot be verified on-chain. For details about batch file format, see [DSNP Spec: Batch Publications](https://spec.dsnp.org/DSNP/BatchPublications).
+
+## Proposal
+* All names are placeholders and may be changed.
+* Errors in the extrinsic(s) must have different, reasonably-named error enums for each type of error for ease of debugging.
+
+### Constants
+* `BatchPageSizeMax` - the maximum number of batch announcements that will be returned by the `get_batches` query
+* `BatchSizeMinBytes` - the minimum possible size of a valid batch file with 1 row
+
+### Enums
+* `BatchFormat` - a list of supported formats for batch files. Currently only Parquet file format is supported, however,
+other formats are being considered.
+
+### Types
+* `BatchAnnouncementParams<T:Config>`: generic
+    * `batch_url`:`Vec<u8>` the URL of the batch file. Must be an IPFS [CIDv1](https://github.com/multiformats/cid/) URL. Accepted codec, algorithm, and base are to be determined.
```

Comment:
Thinking about this one a little more, I'm trying to figure out what is an IPFS CIDv1 URL, and there isn't an official way to do this unless you are talking about using a web gateway, which we aren't.

First, there's no location, so I guess we mean a URI.

Second, the IPFS folks think of this in terms of a path, not a URI.  It's the interplanetary filesystem, after all.  [They don't like `ipfs://` as a scheme specifier.](https://github.com/ipfs/go-ipfs/issues/1678)  They seem to require `foo://ip[f|n]s/<cid>[/<path>]`, where you indicate ipfs or ipns by making it the root of the path, not the scheme.  We don't have to conform to their views.  Some don't.  See, e.g., [Opensea](https://docs.opensea.io/docs/metadata-standards), which adopted `ipfs://`.  I tend toward using `ipfs://`.

I'm assuming there's no reason to support directories or ipns.  And IPNS would open the door to mutable content, which we don't want.  So maybe we can pare this down to just the CID (no path, no root prefix) and maybe we can rename this variable `batch_uri`.  Something like this:

    `batch_uri`:`Vec<u8>`  a URI with an `ipfs` scheme and a path consisting solely of the IPFS CIDv1 of the batch file in multibase format.  Accepted codec, algorithm, and base are to be determined.
    
 This would allow us to expand support to other content-addressed schemas, or even ipfs-based schemas like `nix:///ipfs/<cid>` in the future.

Reply:
I've renamed the field to uri (as you correctly point out), but other details will have to be deferred to when the work is being done.  The revised plan was to just use the CID for now.  I thought I'd put this into another comment but apparently I didn't.